pkg/uprobetracer/tracer.go

```diff
@@ -225,20 +213,20 @@ func (t *Tracer[Event]) attachUprobe(file *os.File) (link.Link, error) {
 // try attaching to a container, will update `containerPid2Inodes`
 func (t *Tracer[Event]) attach(containerPid uint32) {
 	var attachedRealInodes []uint64
-	attachFilePaths, err := t.searchForLibrary(containerPid)
+	unsecuredAttachFilePaths, err := t.searchForLibrary(containerPid)
 	if err != nil {
 		t.logger.Debugf("attaching to container %d: %s", containerPid, err.Error())
 	}
 
-	if len(attachFilePaths) == 0 {
+	if len(unsecuredAttachFilePaths) == 0 {
 		t.logger.Debugf("cannot find file to attach in container %d for symbol %q", containerPid, t.attachSymbol)
 	}
 
-	for _, filePath := range attachFilePaths {
+	for _, filePath := range unsecuredAttachFilePaths {
 		// Do not use `O_PATH` flag here, because `ReadRealInodeFromFd` needs the `private_data` field
 		// in kernel "struct file", to access the underlying inode through overlayFS.
 		// Using `O_PATH` flag will cause the `private_data` field to be zero.
```

Comment:
I think it is still valid.
- `secureopen.OpenInContainer` internally uses `O_PATH` as an implementation detail but returns a file descriptor opened without `O_PATH`.
- `kfilefields.ReadRealInodeFromFd` requires the fd to be opened without `O_PATH`.

I could change the comment to this?

```suggestion
		// Thankfully, OpenInContainer returns a fd opened without `O_PATH`. This
		// is necessary because `ReadRealInodeFromFd` needs the `private_data` field
		// in kernel "struct file", to access the underlying inode through overlayFS.
		// Using `O_PATH` flag will cause the `private_data` field to be zero.
```